### PR TITLE
optimize: add example for Local+BFGS

### PR DESCRIPTION
**Pull request by sbinet**
This CL adds a simple example, extracted from:
 http://docs.scipy.org/doc/scipy-0.17.0/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize

### CODE REVIEW

File: local_example_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/gonum/optimize/functions"
+)
+
+func ExampleLocal_BFGS() {
+	p := optimize.Problem{
+		Func: functions.ExtendedRosenbrock{}.Func,
+		Grad: functions.ExtendedRosenbrock{}.Grad,
+	}
+
+	method := &optimize.BFGS{}
```
Comment (btracey) [inline local_example_test.go]: I'd move this either in the call to local or just before the call to local, that way it's a bit more obvious.

File: local_example_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err = result.Status.Err(); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("result.Status: %v\n", result.Status)
+	fmt.Printf("result.X: %v\n", result.X)
+	fmt.Printf("result.F: %v\n", result.F)
+	// Output:
```
Comment (btracey) [inline local_example_test.go]: Maybe add the number of function evaluations? That's going to be the relevant quantity between methods.

**Comment by btracey** [comment]
LGTM

File: local_example_test.go (hunk 1 above)
Comment (sbinet) [inline local_example_test.go]: done

File: local_example_test.go (hunk 2 above)
Comment (sbinet) [inline local_example_test.go]: done

**Pull request merged**